### ISSUE TITLE
Precompilation needed to use cljlib in a love2d project

### ISSUE THREAD

**Issue by Chouffe** [andreyorst/fennel-cljlib#3]
Thanks a lot for making this library 🤗 It looks amazing and I can't wait to try it out!

I have tried to follow the setup guide to install `cljlib` in my love2d projects. 
I was not able to get it to work and it was complaining about not being able to find the `lazy-seq` module.

If anyone is struggling with the same issue in the future, these are some instructions that made it work for me.
I was able to solve the issue by installing the latest fennel v1.2 and also precompile the fnl files using the command listed in the repository README:

```bash
cd cljlib; COMPILEFLAGS="--require-as-include" make
```

**Comment by andreyorst** [comment]
Interesting. How exactly do you bootstrap love2d? I haven't tried using my libraries with love2d yet, but IIRC when you add `fennel.searcher` to `package.loaders` it should work fine. Maybe it's due to Love's internal file storage, I'm not sure.

Can you upload the minimal project as a zip file?

**Comment by andreyorst** [comment]
Also, I don't think fennel-cljlib really fits for game development, I imagine it being quite slow for that, as I sacrifice performance to provide a similar feature-set of Clojure.

But it can also help find a lot of bottlenecks! :smile:

**Comment by Chouffe** [comment]
Here is my current project: https://github.com/Chouffe/love2d-fennel-breakout-game
I can also create a minimal one that reproduces the issue I encountered.

100% agree with you that it may not be the best lib for the hot paths. With very simple games it may just do and I would trade performance for readability whenever possible :)

**Comment by andreyorst** [comment]
OK, so the issue is with the fact that I've made my own `ns` macro as part of the library, and this macro has the `:require` syntax, reminiscent of Clojure's one, and it makes it such that when you try to load cljlib, cljlib tries to load `lazy-seq` and not `cljlib.lazy-seq`.

This, in turn, means, that only a pre-compiled library will work unless a path is specified for Fennel. So you can run your game like this without pre-compilation of cljlib:

    FENNEL_PATH="cljlib/?/init.fnl;cljlib/itable/src/?.fnl;" love .

I will look into the matter of supporting relative-require for the `ns` macro.

**Comment by andreyorst** [comment]
As of c97f013651a262a62e094b69655bc0adb504dfcc I've added `:relative` as a modifier for `:require` spec, so this should work now.

**Comment by andreyorst** [comment]
@Chouffe I've checked it with your game and it runs, but there might be more cases, so I'll keep this open for now. Feel free to close if everything works

**Comment by andreyorst** [comment]
@Chouffe I've decided to revert this change. Automatic relative require are a hard thing to make correctly, and this change, unfortunately, broke the library in an unexpected way. It was impossible to use the compiled init.lua outside of the cljlib directory.

So the correct way now is to set up the `FENNEL_PATH` and `LUA_PATH` to include the `cljlib` directory or use a compiled `cljlib.lua` file. I've released a new version 1.1.0 and updated the instructions in the README.

Closing this as "works as designed"
